api/envoy/api/v2/route/route_components.proto

```diff
@@ -1072,6 +1086,13 @@ message RetryPolicy {
   repeated HeaderMatcher retriable_request_headers = 10;
 }
 
+message RetryPolicyExtension {
+  // Implementation specific configuration which depends on the implementation being instantiated.
+  // Documentation for :ref:`retry policy extension <arch_overview_http_routing_retry_policy>`. See
+  // the supported retry policy implementations for further documentation.
+  google.protobuf.Any typed_config = 1;
```

Comment:
If we think this is going to be the only field (it probably will be) I would just inline this directly in the message above?